x/wasm/relay_pingpong_test.go

```diff
@@ -0,0 +1,323 @@
+package wasm_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/CosmWasm/go-cosmwasm"
+	wasmTypes "github.com/CosmWasm/go-cosmwasm/types"
+	"github.com/CosmWasm/wasmd/x/wasm"
+	"github.com/CosmWasm/wasmd/x/wasm/ibc_testing"
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/internal/keeper"
+	cosmwasmv2 "github.com/CosmWasm/wasmd/x/wasm/internal/keeper/cosmwasm"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	ping = "ping"
+	pong = "pong"
+)
+const doNotTimeout uint64 = 110000
+
+func TestPinPong(t *testing.T) {
+	var (
+		coordinator = ibc_testing.NewCoordinator(t, 2)
+		chainA      = coordinator.GetChain(ibc_testing.GetChainID(0))
+		chainB      = coordinator.GetChain(ibc_testing.GetChainID(1))
+	)
+	_ = chainB.NewRandomContractInstance() // skip 1 id
+	var (
+		pingContractAddr = chainA.NewRandomContractInstance()
+		pongContractAddr = chainB.NewRandomContractInstance()
+	)
+	require.NotEqual(t, pingContractAddr, pongContractAddr)
+
+	pingContract := &player{t: t, actor: ping, chain: chainA, contractAddr: pingContractAddr}
+	pongContract := &player{t: t, actor: pong, chain: chainB, contractAddr: pongContractAddr}
+
+	wasmkeeper.MockContracts[pingContractAddr.String()] = pingContract
+	wasmkeeper.MockContracts[pongContractAddr.String()] = pongContract
+
+	var (
+		sourcePortID       = wasmkeeper.PortIDForContract(pingContractAddr)
+		counterpartyPortID = wasmkeeper.PortIDForContract(pongContractAddr)
+	)
+	clientA, clientB, connA, connB := coordinator.SetupClientConnections(chainA, chainB, clientexported.Tendermint)
+	connA.NextChannelVersion = ping
+	connB.NextChannelVersion = pong
+
+	channelA, channelB := coordinator.CreateChannel(chainA, chainB, connA, connB, sourcePortID, counterpartyPortID, channeltypes.UNORDERED)
+	var err error
+
+	const startValue uint64 = 100
+	s := startGame{
```

Comment:
After talking, it seems this is what ibc-transfer does. But can we just use the lifetime callback and use startValue 0?

Reply:
Not a blocker if we cannot do this, but it would be nice to set up the channel info on the callback.

Then, the execute message can just contain start value (and max hits?). No need for all source and counterparty, just local channel ID (rest stored from OnIBCChannelConnect)